app/src/main/java/nl/tudelft/b_b_w/controller/BlockController.java

```diff
@@ -67,9 +63,9 @@ public final boolean blockExists(String owner, String key, boolean revoked) {
      * @inheritDoc
      */
     @Override
-    public final List<Block> addBlockToChain(Block block) {
+    public final List<Block> addBlockToChain(Block block) throws HashException {
         // Check if the block already exists
-        String owner = block.getOwner();
+        String owner = block.getOwner().getName();
```

Comment:
Change getOwner() to getUser(). Improves readability a lot. Now I expect when you use getOwner, you'll get a name, which isn't the case.